stdlib/immutableset.gr

```diff
@@ -0,0 +1,499 @@
+/**
+ * @module ImmutableSet: An ImmutableSet is a collection of unique values. Operations on an ImmutableSet do not mutate the set's internal state.
+ * @example import ImmutableSet from "immutableset"
+ * 
+ * @since v0.5.4
+ */
+
+import List from "list"
+import Array from "array"
+
+// implementation based on the paper "Implementing Sets Efficiently in a
+// Functional Language" by Stephen Adams
+
+record Node<a> {
+  key: a,
+  size: Number,
+  left: ImmutableSet<a>,
+  right: ImmutableSet<a>,
+},
+enum ImmutableSet<a> {
```

Comment:
This needs a section comment. We might be waiting on a formatter fix because of the recursive definition?

Reply:
Yeah, it looks like you're right. Should I just include `Node` in the section as well with a TODO comment for when the formatter issue gets fixed?

Reply:
Perhaps you just need a rebase because I believe that was solved with https://github.com/grain-lang/grain/commit/be4b38d87b0ee6ff4544264b8bad4e36820808a2